dessia_common/workflow/blocks.py

```diff
@@ -784,30 +846,57 @@ class SetModelAttribute(Block):
     :param position: Position of the block in canvas.
     """
 
-    def __init__(self, attribute_name: str, name: str = '', position: Tuple[float, float] = None):
-        self.attribute_name = attribute_name
-        inputs = [Variable(name='Model'), Variable(name=f'Value to insert for attribute {attribute_name}')]
-        outputs = [Variable(name=f'Model with changed attribute {attribute_name}')]
+    def __init__(self, attribute_type: AttributeType[Type], name: str = '', position: Tuple[float, float] = None):
+        self.attribute_type = attribute_type
+        parameters = inspect.signature(self.attribute_type.class_).parameters
+        type_ = SetModelAttribute.get_attributes_type(self.attribute_type.name, parameters)
+        if type_:
+            inputs = [TypedVariable(type_= self.attribute_type.class_, name='Model'),
```

Comment:
We can define the first input and the output outside the `if type_:` check

Reply:
Done